frontend/src/book-read/store/apis/review.js

```diff
@@ -0,0 +1,73 @@
+import { createApi } from "@reduxjs/toolkit/query/react";
+import { baseQueryWithReauth } from '../../../auth';
+
+
+export const publishReviewApi = createApi({
+  reducerPath: "publishReviewApi",
+  baseQuery: baseQueryWithReauth,
```

Comment:
There is a bug: when user is not signed in, the GET fetchReviewsByBook request results in unhandled errors (because there is no token to attach) and data not being fetched. 

The source of the problem is within dependency of the whole "publishReviewApi" on the "baseQueryWithReauth". You see, fetchReviewsByBook does not require authentication, and yet authentication is forced upon it by base query.

One way to fix this bug is to write a local function that will return either baseQuery or baseQueryWithReauth depending on the parameters of the request (e.g. URL).

Here is a (not entirely) related example:
https://stackoverflow.com/questions/69568818/how-to-dynamically-change-base-url-using-redux-toolkit
Out of the passed values you will probably need "args". But I don't know what is contained in it. You can check documentation or use debug to find out.

Reply:
Fixed the issue, also had to update the Drawer structure to remove the extra network request